components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -1089,6 +1101,9 @@ private List<ApplicationListItem> getApplicationListItems(List<ServiceProvider>
             if (requiredAttributes.stream().noneMatch(attribute -> attribute.equals(ADVANCED_CONFIGURATIONS))) {
                 applicationResponseModel.advancedConfigurations(null);
             }
+            if (requiredAttributes.stream().noneMatch(attribute -> attribute.equals(CLIENT_ID))) {
```

Comment:
Double check whether there are cases like retrieving clientId from framework side although the clientId is not in the requiredAttributes list.